File: Back-End/SmartAttendance.DAL/DBModels/InstructorCourse.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;


namespace SmartAttendance.DAL.DBModels
{
    public class InstructorCourse
    {
        [Key]
        public int Id { get; set; }
        public int InstructorId { get; set; }
        public double TotalAttendanceGrade { get; set; }
        public string Term { get; set; }
        public DateTime CreatedAt { get; set; }
        public int CourseId { get; set; }
        [ForeignKey("InstructorId")]
        public User Instructor { get; set; }
        [ForeignKey("CourseId")]
        public Course Course { get; set; }
        public ICollection<Lucture>? Lectures { get; set; }
    }
}

File: Front/SmartAttendance/src/app/SmartAttendance/Model/instructor-course.ts
import { Lucture } from "./lucture";

export class InstructorCourse {
    public  id :number = 0;
    public  instructorId :number = 0;
    public  totalAttendanceGrade :number = 0;
    public  term :string = '';
    public  createdAt :string = '';
    public  courseId :number = 0;
    public  courseName :string = '';
    public  departmentName :string = '';
    public  departmentId :number = 0;
    public  lectures  :Lucture[] = []
}
